utilities_test.go

```diff
@@ -96,7 +97,86 @@ func TestDatasetWarp(t *testing.T) {
 		t.Errorf("wrong block size %d,%d", st.BlockSizeX, st.BlockSizeY)
 	}
 }
+func TestDatasetWarpMulti(t *testing.T) {
+	ds1, _ := Open("testdata/test_1_1.tif", RasterOnly())
+	ds2, _ := Open("testdata/test_1_2.tif", RasterOnly())
+	ds3, _ := Open("testdata/test_2_1.tif", RasterOnly())
+	ds4, _ := Open("testdata/test_2_2.tif", RasterOnly())
 
+	defer ds1.Close()
+	defer ds2.Close()
+	defer ds3.Close()
+	defer ds4.Close()
+
+	// Warp NewDataset with multiple input dataset
+	filePath := path.Join(os.TempDir(), "warp.tif")
+	outputDataset, err := Warp(filePath, []*Dataset{ds1, ds2, ds3, ds4}, []string{"-ts", "40", "40"}, CreationOption("TILED=YES", "BLOCKXSIZE=32", "BLOCKYSIZE=16"), GTiff)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer os.Remove(filePath)
+	defer outputDataset.Close()
+
+	st := outputDataset.Structure()
```

Comment:
we don't need to test structure or geotransform here, the warp options/switches are already covered n other tests